pkg/storage/cloud/cloud_io.go

```diff
@@ -80,13 +83,22 @@ const MaxDelayedRetryAttempts = 3
 // fails. It knows about specific kinds of errors that need longer retry
 // delays than normal.
 func DelayedRetry(
-	ctx context.Context, customDelay func(error) time.Duration, fn func() error,
+	ctx context.Context, opName string, customDelay func(error) time.Duration, fn func() error,
 ) error {
+	span := tracing.SpanFromContext(ctx)
+	attemptNumber := int32(1)
 	return retry.WithMaxAttempts(ctx, base.DefaultRetryOptions(), MaxDelayedRetryAttempts, func() error {
 		err := fn()
 		if err == nil {
 			return nil
 		}
+		redactedErr := string(redact.Sprintf("%v", err))
+		retryEvent := &roachpb.RetryTracingEvent{
+			Operation:     opName,
+			AttemptNumber: attemptNumber,
+			RetryError:    redactedErr[:100], // truncate the error string
```

Comment:
I'd prefer to have this redaction + truncation pulled out to a helper method. At the very least it would keep the truncation length in 1 place. 100 also seems a bit low, but I don't have much backing that claim... maybe 250 similarly arbitrarily?

Reply:
done.